### PR TITLE
Always set standard security headers for OAuth server

### PR DESCRIPTION
**Pull request by enj**
We should always set these headers on all pages of the OAuth server.

Signed-off-by: Monis Khan <mkhan@redhat.com>

@openshift/sig-auth

### CODE REVIEW

File: test/integration/oauth_server_headers_test.go [hunk 1]
```diff
@@ -95,19 +113,30 @@ func checkNewReqHeaders(t *testing.T, rt http.RoundTripper, checkUrl string) {
...
 	// only add items to this list if they cannot be statically checked above
-	ignoredHeaders := []string{"Audit-Id", "Date", "Content-Length", "Location"}
+	ignoredHeaders := []string{"Audit-Id", "Date", "Content-Type", "Content-Length", "Location"}
 	for _, h := range ignoredHeaders {
 		resp.Header.Del(h)
 	}
 
+	// tolerate additional header set by osin library code
+	expires := resp.Header["Expires"]
+	if len(expires) == 2 && expires[1] == "Fri, 01 Jan 1990 00:00:00 GMT" {
```
Comment (ericavonb) [inline test/integration/oauth_server_headers_test.go]: wat? I know osin library is crazy but what's going on here?

Comment (enj) [reply test/integration/oauth_server_headers_test.go]: It insists on blindly adding those headers -.-

**Pull request merged**